homeassistant/helpers/typing.py

```diff
@@ -31,3 +40,12 @@ class UndefinedType(Enum):
 
 
 UNDEFINED = UndefinedType._singleton  # pylint: disable=protected-access
+
+
+class AddEntitiesCallback(Protocol):
```

Comment:
Creating a class (with the overhead it has) to provide type-checking for a parameter where the type is optional seems like a lot.

Reply:
What overhead it has?

Reply:
You could achieve the same with an alias (see start of typing.py) which is just a text replacement and have no runtime effect.

Reply:
Alias won't let to define optional or named argument. Also aliases are actually evaluated during runtime, that's why we need use `Dict` there and not modern `dict`.

This has no negative runtime effects.

Reply:
Let us not discuss, I have given my opinion and others will give theirs.

Reply:
I think a user defined protocol is fitting in this case.
Mainly as it nicely exposes the keyword arguments.

Reply:
Yes, with `Protocol` it's now possible to write `async_add_entities(entities, update_before_add = True)` and mypy will validate this call.